persistence/data_models_json_fields.go

```diff
@@ -253,12 +256,13 @@ func (s *StateExecutionLocalQueuesJson) ConsumeWithCheckingLocalQueueWaitingComp
 		}
 	}
 
-	if idx < len(s.StateToLocalQueueCommandsMap[stateExecutionIdKey]) {
-		remainingCommands = append(remainingCommands, s.StateToLocalQueueCommandsMap[stateExecutionIdKey][idx+1:]...)
+	for idx+1 < len(s.StateToLocalQueueCommandsMap[stateExecutionIdKey]) {
+		idx += 1
+		remainingCommands[idx] = s.StateToLocalQueueCommandsMap[stateExecutionIdKey][idx]
```

Comment:
I guess this line should be before idx+=1? 
otherwise the idx could be out of index

Reply:
The line 259 is `for idx+1 < len(s.StateToLocalQueueCommandsMap[stateExecutionIdKey])`, so idx + 1 is always in the index.